pkg/dogstatsd/server.go

```diff
@@ -376,6 +415,7 @@ func NewServer(demultiplexer aggregator.Demultiplexer, serverless bool) (*Server
 			serverlessMode:            serverless,
 			originOptOutEnabled:       config.Datadog.GetBool("dogstatsd_origin_optout_enabled"),
 		},
+		Logger: logger,
```

Comment:
Lets call this `DogstatsdDebugLogger` (or similar) so a future reader knows what its for.